servers/physics_2d/broad_phase_2d_hash_grid.cpp

```diff
@@ -35,6 +35,12 @@
 #define LARGE_ELEMENT_FI 1.01239812
 
 void BroadPhase2DHashGrid::_pair_attempt(Element *p_elem, Element *p_with) {
+	if (!_test_collision_mask(p_elem->collision_mask, p_elem->collision_layer, p_with->collision_mask, p_with->collision_layer)) {
+		return;
+	}
+	if (p_elem->owner == p_with->owner) {
+		return;
+	}
```

Comment:
Checking that it's not the same owner could go before testing the collision layer to avoid unnecessary operations in common cases.